packages/vite/src/node/plugins/asset.ts

```diff
@@ -470,15 +477,23 @@ async function fileToBuiltUrl(
       map.set(contentHash, fileName)
     }
     const emittedSet = emittedHashMap.get(config)!
+    const name = normalizePath(path.relative(config.root, file))
     if (!emittedSet.has(contentHash)) {
-      const name = normalizePath(path.relative(config.root, file))
       pluginContext.emitFile({
         name,
         fileName,
         type: 'asset',
         source: content
       })
       emittedSet.add(contentHash)
+    } else {
+      duplicateAssets.push({
+        name,
+        fileName,
+        type: 'asset',
+        source: content,
+        isAsset: true
+      })
```

Comment:
Instead of working around the hashing problem, I think we should fix it by hashing the filename instead of the content (or simply use the filename without hashing). It would cause Vite 3.1 to generate a different set of hashes, but it feels like the right thing to do here.

---

Would be great to have test for this too 🙂

Reply:
So you think the dedupe wasnt intentional? I thought we may need a team discusion to change it. See tge linked issue, my take is that we shouldnt dedupe but it feels like a v4 change to me

Reply:
It's probably intentional from https://github.com/vitejs/vite/commit/f8e4eebcc48ad5e98e52d12ce6595da014a43276, but only Evan knows 😬  And judging from that commit which references https://github.com/rollup/rollup/issues/3415, it'll be fixed in Rollup 3 too.

I think a team discussion make sense just to be safe, but I saw this PR marked to close #8632 so was a bit cautious. I think not deduping would be ideal too and wouldn't actually be breaking stuff.

Reply:
So, lets move forward with this for 3.1 so we unblock laravel. And we can later discuss with the team.

Reply:
Alright, at the meantime we can try to fix the manifest first then. I'll put that issue on the team board.